### ISSUE TITLE
Bug: malloc error

### ISSUE THREAD

**Issue by IkerGalardi** [IkerGalardi/LMake#69]
Platform info:
- **OS**: arch linux
- **KERNEL**: 5.4.82-1-lts

When executing `lmake build` the next error appears:
`lmake: malloc.c:2394: sysmalloc: Assertion (old_top == initial_top (av) && old_size == 0) || ((unsigned long) (old_size) >= MINSIZE && prev_inuse (old_top) && ((unsigned long) old_end & (pagesize - 1)) == 0)' failed.
[1]    10489 abort (core dumped)  lmake build` 

That error occurs when linking. The project with this error happens on branch [lmake](https://github.com/IkerGalardi/LMake/tree/lmake)

**Comment by IkerGalardi** [comment]
The error has stopped appearing, won't close the issue because the bug can still be there.

**Comment by IkerGalardi** [comment]
Error seems to come from function `lmake_link` file.

**Comment by IkerGalardi** [comment]
Output given by the debug flag:

```
lmake build --debug
[D] Changing the directory to /home/iquer/Projects/LMake/lmakebuild
[D] Finding recursivelly with regex src/**.cc
lmake: malloc.c:2394: sysmalloc: Assertion `(old_top == initial_top (av) && old_size == 0) || ((unsigned long) (old_size) >= MINSIZE && prev_inuse (old_top) && ((unsigned long) old_end & (pagesize - 1)) == 0)' failed.
[1]    17741 abort (core dumped)  lmake build --debug

```

**Comment by IkerGalardi** [comment]
Merged in commit [7ff32dea18de3b97e7a8c8c05f5f97fbcad5f24b](https://github.com/IkerGalardi/LMake/tree/7ff32dea18de3b97e7a8c8c05f5f97fbcad5f24b)
